From cff1dfd767d1ee3c773fd8b57fe310957e5f8abb Mon Sep 17 00:00:00 2001
From: Theodore Ts'o <tytso@mit.edu>
Date: Fri, 9 Sep 2011 19:12:51 -0400
Subject: [PATCH 21/26] ext4: rename ext4_free_blocks_after_init() to ext4_free_clusters_after_init()
Patch-mainline: 3.2

This function really returns the number of clusters after initializing
an uninitalized block bitmap has been initialized.

Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
Signed-off-by: Robin Dong <sanbai@taobao.com>
---
Index: linux-2.6.32-279.2.1.el5/fs/ext4/balloc.c
===================================================================
--- linux-2.6.32-279.2.1.el5.orig/fs/ext4/balloc.c
+++ linux-2.6.32-279.2.1.el5/fs/ext4/balloc.c
@@ -210,9 +210,9 @@ void ext4_init_block_bitmap(struct super
 /* Return the number of free blocks in a block group.  It is used when
  * the block bitmap is uninitialized, so we can't just count the bits
  * in the bitmap. */
-unsigned ext4_free_blocks_after_init(struct super_block *sb,
-				     ext4_group_t block_group,
-				     struct ext4_group_desc *gdp)
+unsigned ext4_free_clusters_after_init(struct super_block *sb,
+				       ext4_group_t block_group,
+				       struct ext4_group_desc *gdp)
 {
 	return num_clusters_in_group(sb, block_group) -
 		ext4_num_overhead_clusters(sb, block_group, gdp);
Index: linux-2.6.32-279.2.1.el5/fs/ext4/ext4.h
===================================================================
--- linux-2.6.32-279.2.1.el5.orig/fs/ext4/ext4.h
+++ linux-2.6.32-279.2.1.el5/fs/ext4/ext4.h
@@ -1643,9 +1643,9 @@ extern void ext4_init_block_bitmap(struc
 				   struct buffer_head *bh,
 				   ext4_group_t group,
 				   struct ext4_group_desc *desc);
-extern unsigned ext4_free_blocks_after_init(struct super_block *sb,
-					    ext4_group_t block_group,
-					    struct ext4_group_desc *gdp);
+extern unsigned ext4_free_clusters_after_init(struct super_block *sb,
+					      ext4_group_t block_group,
+					      struct ext4_group_desc *gdp);
 extern unsigned ext4_num_base_meta_clusters(struct super_block *sb,
 					    ext4_group_t block_group);
 extern unsigned ext4_num_overhead_clusters(struct super_block *sb,
Index: linux-2.6.32-279.2.1.el5/fs/ext4/ialloc.c
===================================================================
--- linux-2.6.32-279.2.1.el5.orig/fs/ext4/ialloc.c
+++ linux-2.6.32-279.2.1.el5/fs/ext4/ialloc.c
@@ -967,7 +967,7 @@ got:
 		if (gdp->bg_flags & cpu_to_le16(EXT4_BG_BLOCK_UNINIT)) {
 			gdp->bg_flags &= cpu_to_le16(~EXT4_BG_BLOCK_UNINIT);
 			ext4_free_group_clusters_set(sb, gdp,
-				ext4_free_blocks_after_init(sb, group, gdp));
+				ext4_free_clusters_after_init(sb, group, gdp));
 			gdp->bg_checksum = ext4_group_desc_csum(sbi, group,
 								gdp);
 		}
Index: linux-2.6.32-279.2.1.el5/fs/ext4/mballoc.c
===================================================================
--- linux-2.6.32-279.2.1.el5.orig/fs/ext4/mballoc.c
+++ linux-2.6.32-279.2.1.el5/fs/ext4/mballoc.c
@@ -2247,7 +2247,7 @@ int ext4_mb_add_groupinfo(struct super_b
 	 */
 	if (desc->bg_flags & cpu_to_le16(EXT4_BG_BLOCK_UNINIT)) {
 		meta_group_info[i]->bb_free =
-			ext4_free_blocks_after_init(sb, group, desc);
+			ext4_free_clusters_after_init(sb, group, desc);
 	} else {
 		meta_group_info[i]->bb_free =
 			ext4_free_group_clusters(sb, desc);
@@ -2825,7 +2825,7 @@ ext4_mb_mark_diskspace_used(struct ext4_
 	if (gdp->bg_flags & cpu_to_le16(EXT4_BG_BLOCK_UNINIT)) {
 		gdp->bg_flags &= cpu_to_le16(~EXT4_BG_BLOCK_UNINIT);
 		ext4_free_group_clusters_set(sb, gdp,
-					     ext4_free_blocks_after_init(sb,
+					     ext4_free_clusters_after_init(sb,
 						ac->ac_b_ex.fe_group, gdp));
 	}
 	len = ext4_free_group_clusters(sb, gdp) - ac->ac_b_ex.fe_len;
